src/lib/flows/common/build-chain-flow-helper.js

```diff
@@ -207,37 +207,37 @@ async function checkoutProjectPullRequestFlow(
 async function checkoutNode(context, node, checkoutInfo, dir) {
   if (checkoutInfo.merge) {
     logger.info(
-      `[${node.project}] Rebasing ${context.config.github.serverUrl}/${checkoutInfo.group}/${checkoutInfo.project}:${checkoutInfo.branch} onto ${context.config.github.serverUrl}/${node.project}:${checkoutInfo.targetBranch}`
+      `[${node.project}] Merging ${context.config.github.serverUrl}/${node.project}:${checkoutInfo.targetBranch} into ${context.config.github.serverUrl}/${checkoutInfo.group}/${checkoutInfo.project}:${checkoutInfo.branch}`
     );
     try {
       await clone(
-        `${context.config.github.serverUrlWithToken}/${checkoutInfo.group}/${checkoutInfo.project}`,
+        `${context.config.github.serverUrlWithToken}/${node.project}`,
         dir,
-        checkoutInfo.branch
+        checkoutInfo.targetBranch
       );
     } catch (err) {
       logger.error(
-        `[${node.project}] Error checking out (before rebasing) ${context.config.github.serverUrl}/${checkoutInfo.group}/${checkoutInfo.project}:${checkoutInfo.branch}`
+        `[${node.project}] Error checking out (before merging) ${context.config.github.serverUrl}/${node.repo.group}/${node.project}:${context.config.github.targetBranch}`
       );
       throw err;
     }
     try {
-      await fetchFromRemote(
+      await gitMerge(
         dir,
-        `${context.config.github.serverUrlWithToken}/${node.project}`,
-        checkoutInfo.targetBranch
+        `${context.config.github.serverUrlWithToken}/${checkoutInfo.group}/${checkoutInfo.project}`,
+        checkoutInfo.branch
       );
     } catch (err) {
       logger.error(
-        `[${node.project}] Error fetching target branch (before rebasing) ${context.config.github.serverUrl}/${node.project}:${checkoutInfo.targetBranch}`
+        `[${node.project}] Error merging ${context.config.github.serverUrl}/${checkoutInfo.group}/${checkoutInfo.project}:${checkoutInfo.branch}. Please manually merge it and relaunch.`
       );
       throw err;
     }
     try {
-      await rebase(dir, checkoutInfo.targetBranch);
+      await gitRename(dir, checkoutInfo.branch);
```

Comment:
just want to understand the need for renaming it :thinking: . Can you please clarify?

Reply:
Originally the rebasing was introduced to fix the Sonar zero issues problem. This happened because the base branch used for merging is the base one, `main`, `7.x`, etc. The solution to this is to keep the result of the branch with the name of the source of the pull request.

Reply:
thanks @almope